soc/nuvoton/npcx/common/scfg.c

```diff
@@ -105,6 +105,35 @@ void npcx_pinctrl_i2c_port_sel(int controller, int port)
 	}
 }
 
+void npcx_i2c_target_start_wk_enable(int controller, bool enable)
+{
+	struct glue_reg *const inst_glue = HAL_GLUE_INST();
+
+	if (enable) {
+		/* Clear Start condition detection status */
+		inst_glue->SMB_SBD |= BIT(controller);
+		/* Enable wake up event assertion */
+		inst_glue->SMB_EEN |= BIT(controller);
+	} else {
+		/* Disable wake up event assertion */
+		inst_glue->SMB_EEN &= ~BIT(controller);
+	}
+}
+
+void npcx_i2c_target_clear_detection_event(void)
+{
+	struct glue_reg *const inst_glue = HAL_GLUE_INST();
+	uint8_t een = inst_glue->SMB_EEN;
+	uint8_t sbd = inst_glue->SMB_SBD;
+
+	/* Clear Start condition detection status */
+	for (int i = 0; i < 8; i++) {
```

Comment:
`uint8_t`, save few bytes on the stack (maybe)

Reply:
Done